stock-tracker/src/features/chart/redux/chartReducer.ts

```diff
@@ -20,7 +20,7 @@ export const initialState: ChartState = {
   error: false
 };
 
-export const chartReducer: Reducer<ChartState, ChartActions> = (
+export const chartReducer: Reducer<Readonly<ChartState>, ChartActions> = (
```

Comment:
Why use `Readonly` here?

Reply:
Readonly makes sure that developers cannot mutate the state within the reducer therefore ensuring immutability.